operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/NetworkPolicySpec.java

```diff
@@ -17,23 +17,45 @@
 
 package org.keycloak.operator.crds.v2alpha1.deployment.spec;
 
+import java.util.List;
 import java.util.Optional;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import io.fabric8.kubernetes.api.model.networking.v1.NetworkPolicyPeer;
 import io.sundr.builder.annotations.Buildable;
 import org.keycloak.operator.Constants;
+import org.keycloak.operator.crds.v2alpha1.CRDUtils;
 import org.keycloak.operator.crds.v2alpha1.deployment.Keycloak;
+import org.keycloak.operator.crds.v2alpha1.deployment.KeycloakSpec;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @Buildable(editableEnabled = false, builderPackage = "io.fabric8.kubernetes.api.builder")
 public class NetworkPolicySpec {
 
+    // Copied from Kubernetes Documentation
+    private static final String RULE_DESCRIPTION = "A list of sources which should be able to access this endpoint. " +
+            "Items in this list are combined using a logical OR operation. " +
+            "If this field is empty or missing, this rule matches all sources (traffic not restricted by source). " +
+            "If this field is present and contains at least one item, this rule allows traffic only if the traffic matches at least one item in the from list.";
+
     @JsonProperty("enabled")
     @JsonPropertyDescription("Enables or disable the ingress traffic control.")
     private boolean networkPolicyEnabled = false;
 
+    @JsonProperty("http")
+    @JsonPropertyDescription(RULE_DESCRIPTION)
+    private List<NetworkPolicyPeer> httpRules;
+
+    @JsonProperty("https")
+    @JsonPropertyDescription(RULE_DESCRIPTION)
+    private List<NetworkPolicyPeer> httpsRules;
```

Comment:
No worries. My reasoning for keeping both separated is flexibility. I'm not sure if it is a common use case to have both HTTP and HTTPS enabled at the same time, but it is flexible for those scenarios.

Reply:
+1 for keeping both as people might have both port, and then want to configure both separately. Otherwise we need to "fix" it first that they can have either http or https enabled :smile: